CONTRIBUTORS.md

```diff
@@ -0,0 +1,118 @@
+
+* **Improving documentation:** py4DSTEM's documentation is always in need of improvement. If you have suggestions for improving the documentation, please submit a pull request or open an issue on GitHub.
+
+* **Developing new code:** If you are a developer, you can contribute to py4DSTEM by writing new code. Please follow the coding guidelines below.
+
+### Coding Guidelines
+
+* **Code style:** py4DSTEM uses the black code formatter and flake8 linter. All code must pass these checks without error before it can be merged. We suggest using `pre-commit` to help ensure any code commited follows these practices, checkout the [setting up developer environment section below](#install). We also try to abide by PEP8 coding style guide where possible.
+
+* **Documentation:** All code should be well-documented, and use Numpy style docstrings. Use docstrings to document functions and classes, add comments to explain complex code both blocks and individual lines, and use informative variable names.
+
+* **Testing:** Ideally all new code should be accompanied by tests using pyTest framework; at the least we require examples of old and new behaviour caused by the PR. For bug fixes this can be a block of code which currently fails and works with the proposed changes. For new workflows or extensive feature additions, please also include a Jupyter notebook demonstrating the changes for an entire workflow i.e. from loading the input data to visualizing and saving any processed results. 
+
+* **Dependencies:** New dependencies represent a significant change to the package, and any PRs which add new dependencies will require discussion and agreement from the development team. If a new dependency is required, please prioritize adding dependencies that are actively maintained, have permissive installation requirements, and are accessible through both pip and conda.
+
+### Pull Requests
+
+When submitting a pull request, please:
+
+* **Open a corresponding issue:** Before submitting a pull request, open an issue to discuss your changes and get feedback.
+
+* **Write a clear and concise pull request description:** The pull request description should clearly explain the changes you made and why they are necessary.
+
+* **Follow the coding guidelines:** Make sure your code follows the coding guidelines outlined above.
+
+* **Add tests:** If your pull request includes new code, please add unit tests.
+
+* **Address reviewer feedback:** Respond promptly to reviewer feedback and make changes as needed.
+
+### Code Review
+
+All pull requests will be reviewed by project maintainers before they are merged. The maintainers will provide feedback on the code and may ask for changes. Please be respectful of the maintainers' feedback and make changes as needed.
+
+### Code of Conduct
+
+py4DSTEM is committed to providing a welcoming and inclusive environment for all contributors. Please read and follow the py4DSTEM Code of Conduct.
```

Comment:
If we're going to reference this we probably need to add it!!

Reply:
added